### PR TITLE
Initial services app flow

### PR DESCRIPTION
**Pull request by davydka**
Initial setup for the `Services` flow.
Created routes and modified copy, styles, and app logic for:
* ENTRY
* EMAIL_FORM
* EMAIL_SENT
* EMAIL_CONFIRM (not in designs, but used for logging in)
* CONTACT_FORM
* CONTACT_CONFIRMATION
* SERVICE_TYPE

Large GIF (376x814)
Large GIF (376x814)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -102,6 +109,37 @@ function App({ backend }) {
...
+              <ContactConfirmation backend={backend} />
+            </ProtectedRoute>
+            <ProtectedRoute backend={backend} exact path={Routes.SERVICE_TYPE}>
+              <ServiceType backend={backend} />
+            </ProtectedRoute>
+            <Route path="*">
+              <NoMatch />
+            </Route>
+            {/*END MVP Routes*/}
+
```
Comment (davydka) [inline src/App.js:142]: All of the new routes live here.

File: src/App.js [hunk 2]
```diff
@@ -23,10 +23,17 @@ import NewFacility from 'pages/facility_new';
...
+// MVP
+import EntryPortal from 'pages/entry';
 import SignUp from 'pages/signup';
 import SignUpConfirmation from 'pages/signup_confirmation';
+import SignupFinish from 'pages/signup_approve';
+import Contact from 'pages/contact';
+import ContactConfirmation from 'pages/contact_confirmation';
+import ServiceType from 'pages/service_type';
+// End MVP
```
Comment (davydka) [inline src/App.js:36]: Currently active pages being used.

File: src/components/Form/FormBuilder.js [hunk 3]
```diff
@@ -19,11 +18,9 @@ function FormBuilder({
   onSubmit,
   title,
 }) {
-  const { t } = useTranslation();
-
   return (
     <Form
-      buttonLabel={buttonLabel || t('generic.form.submitLabel')}
```
Comment (lizbanach) [inline src/components/Form/FormBuilder.js:26]: just wondering - why was this or value removed?

File: src/components/SignupFinish.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+import { Routes } from 'constants/Routes';
+import { routeWithParams } from 'lib/utils/routes';
+import { isValidEmail } from 'lib/utils/validations';
+
+import { useAuth } from 'hooks/useAuth';
+
+import PageLoader from 'components/Loader/PageLoader';
+import Note from 'components/Note';
+import Anchor, { anchorTypes } from 'components/Anchor';
```
Comment (lizbanach) [inline src/components/SignupFinish.js:19]: this is a nit - but can we organize imports by type?

external libs
internal libs
hooks
components
components in relative path (./blah)

File: src/components/Form/FormBuilder.js [hunk 5]
```diff
@@ -19,11 +18,9 @@ function FormBuilder({
...
+      buttonLabel={buttonLabel}
```
Comment (davydka) [inline src/components/Form/FormBuilder.js:23]: `Form` component has a new default behavior for `Submit` button. Setting the `buttonLabel` overrides that default behavior, so taking this label out of the Formbuilder allows the buttonLabel override to tunnel down into the Form.

File: src/components/layouts/Page/index.js [hunk 6]
```diff
@@ -33,10 +40,12 @@ const PageContent = ({
 
 const Page = ({
   children,
+  contentContainerStyles,
   hasBackButton = true,
   isHome,
   onBackButtonClick,
   currentProgress,
+  rootContainerStyles,
```
Comment (davydka) [inline src/components/layouts/Page/index.js:48]: Adding style overrides for the Page container and the content container within the page.

File: src/constants/Routes.js [hunk 7]
```diff
@@ -21,4 +21,12 @@ export const Routes = {
...
   SUPPLY_NEW_ADMIN: `/new/admin/supply/:id`,
   SUPPLY_NEW_ADMIN_CONFIRMATION: `/new/admin/supply/:id/confirmation/:requestId`,
+
+  // MVP Routes
+  EMAIL_FORM: `/signin`,
+  EMAIL_SENT: `/signin/confirm`,
+  EMAIL_APPROVE: `/signin/complete`,
+  CONTACT_FORM: `/contact`,
+  CONTACT_CONFIRMATION: `/contact/confirm`,
+  SERVICE_TYPE: '/service',
```
Comment (davydka) [inline src/constants/Routes.js:31]: New routes.

File: src/containers/ServiceTypeForm.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        { label: 'Emotional support', value: 'emotional-support' },
+      ],
+      name: 'serviceType',
+      type: formFieldTypes.INPUT_DROPDOWN,
+      value: fields.serviceType,
+    },
+    {
+      customOnChange: handleFieldChange('urgency'),
+      label: t('service.selectType.form.urgency.label'),
+      // todo: move these values to enum
```
Comment (davydka) [inline src/containers/ServiceTypeForm.js:59]: 🙏 next steps.

File: src/pages/contact.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import ServiceContactForm from 'containers/ServiceContactForm';
+
+function Contact({ backend }) {
+  const history = useHistory();
+  const [isLoading] = useState(false);
+
+  return (
+    <Page
+      onBackButtonClick={() =>
+        // todo: if user has contact info set, route to userDetail page
```
Comment (davydka) [inline src/pages/contact.js:19]: `userDetail` page hasn't been built yet.

File: src/pages/contact_confirmation.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+import { Routes } from 'constants/Routes';
+import { routeWithParams } from 'lib/utils/routes';
+
+import Page from 'components/layouts/Page';
+import PageLoader from 'components/Loader/PageLoader';
+import { ContactConfirmation } from 'components/Confirmation';
+
+const TEMP_NAME = 'Temp User';
+const TEMP_CONTACT = 'medicalWorker@healthcarefacility.com';
```
Comment (davydka) [inline src/pages/contact_confirmation.js:14]: Temp values while we build backend for this part.

Comment (lizbanach) [reply src/pages/contact_confirmation.js:14]: could you add a note above this?

**Comment by lizbanach** [comment]
this is a great start! few nits and questions - also, i noticed that the submit button label is borked:

Screen Shot 2020-04-10 at 12 21 19 PM
Screen Shot 2020-04-10 at 12 21 53 PM

File: src/components/Form/FormBuilder.js (hunk 3 above)
Comment (davydka) [reply src/components/Form/FormBuilder.js:26]: Totally!
The `Form` component has a new default behavior for Submit button. Setting the `buttonLabel` overrides that default behavior, so taking this label out of the Formbuilder allows the `buttonLabel` override to tunnel down into the Form.

Comment (davydka) [reply src/components/Form/FormBuilder.js:26]: Here's the logic for the Submit button label:
https://github.com/helpsupply/helpsupply/blob/3c19497980883240cb5c6ae3cca182a9ff409755/src/components/Form/index.js#L48

**Pull request merged**